### ISSUE TITLE
ERR_SPDY_PROTOCOL_ERROR in Chrome on External/Callback

### ISSUE THREAD

**Issue by egulbrand** [IdentityServer/IdentityServer4#2735]
Hello,
After deploying my implementation of IdentityServer4 to IIS, everything seems to work okay except for the External/Callback route.  I can get access tokens using the client credentials flow.  When I attempt to authenticate with an external provider though (ADFS) and the browser is redirected back to the IdentityServer, I get this ERR_SPDY_PROTOCOL_ERROR issue in Chrome.  If I try this in Firefox, I get a 'The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.' message.  I am using a third party wildcard SSL certificate from GoDaddy that seems to work for other sites I have.  Any ideas as to why I would be getting this error on External/Callback? Or can I get some guidance on logging that may give me an idea as to what is going wrong?

Any assistance is appreciated. Thanks.

**Comment by ghanashyaml** [comment]
Could be cause by Adblock. See here:
https://www.itechgyan.com/err_spdy_protocol_error/

**Comment by egulbrand** [comment]
After doing more digging into this, @brockallen is correct.  The ADFS external provider I was calling was configured against an AWS Directory Service.  After debugging the ExternalController, Callback method, I noticed that AWS puts an Active Directory user into many groups by default.  In our case, 21 groups.  In the end there was a total of 36 claims about the user.  I had to add some registry entries on the server for IIS to allow for the increased length.  The registry entries were added at HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\HTTP\Parameters.  The entries were:

MaxFieldLength: 65534 
MaxRequestBytes: 16777216 
UriMaxUriBytes: 16777216 
UrlSegmentMaxLength: 32766

I then had to restart the server.  'IISRESET' did not apply these changes.  It worked after restarting the server.

I found this solution at [http://tjluis.blogspot.com/2016/06/solved-url-too-long-on-iis-for-good.html](url)
